build.sbt

```diff
@@ -0,0 +1,41 @@
+import sbt.Keys.version
+
+name := {
+  println(s"[info] Welcome to scalagen ${version.value}")
+  "scalagen"
+}
+
+lazy val sharedSettings = Def.settings(
+  updateOptions := updateOptions.value.withCachedResolution(true),
+  organization := "org.scalameta",
+  version := "0.1",
+  scalaVersion := "2.12.4",
+  libraryDependencies ++=
+    "org.scalameta" %%% "scalameta" % "2.1.3" ::
+      "org.scalameta" %%% "contrib" % "2.1.3" ::
+      "org.scalactic" %%% "scalactic" % "3.0.4" ::
+      "org.scalatest" %%% "scalatest" % "3.0.4" % "test" :: Nil,
+  scalacOptions ++= "-Xfatal-warnings" :: Nil
+)
+
+lazy val scalagen =
+  crossProject
+    .in(file("scalagen"))
+    .settings(sharedSettings)
+
+lazy val scalagenJVM = scalagen.jvm
+lazy val scalagenJS = scalagen.js
+
+// JVM sbt plugin
+lazy val sbtScalagen =
+  project
+    .in(file("sbt-scalagen"))
+    .settings(sharedSettings)
+    .dependsOn(scalagenJVM)
+
+// JVM based cli. Could be ported to SN instead
+lazy val cliScalagen =
+  project
+    .in(file("cli-scalagen"))
```

Comment:
Conventionally this should be `scalagen-cli`, this makes the directories align nicely like in https://github.com/scalacenter/scalafix I think this is an inherited convention from maven https://github.com/vsch/flexmark-java

Only sbt plugins that should conventionally have `moduleName := "sbt-scalagen"` but the code should live in `scalagen-sbt`

Reply:
Sounds good